### PR TITLE
Allow service and deployment specific annotations

### PR DESCRIPTION
**Pull request by derrickburns**
Addresses #35

### CODE REVIEW

**Comment by travisgroth** [comment]
Hi @derrickburns - I think the logic to have per-service overrides is easier to follow if you do something like:

```
  annotations:
{{- if .Values.authenticate.deployment.annotations }}
  {{- range $key, $value := .Values.authenticate.deployment.annotations }}
    {{ $key }}: {{ $value | quote }}
  {{- end }}
{{- else if .Values.annotations }}
  {{- range $key, $value := .Values.annotations }}
    {{ $key }}: {{ $value | quote }}
  {{- end }}
{{- end }}
``` 

You can do a similar approach with `service.annotations`.  This is a slightly more intrusive, but cleaner end state.

Please also be sure to sign the CLA, version bump the chart, and update the README as well.  Heads up - we mirror our chart in `helm/stable`, and there is probably an inbound merge to rebase on shortly.

**Comment by derrickburns** [comment]
I have tried to execute the CLA several times.  Giving up.

**Comment by desimone** [comment]
Sorry for the frustration. 

I’m fine with DCO. I’m going to turn off CLA for all but main repo. Thanks for the PR

**Comment by travisgroth** [comment]
Hey @derrickburns thanks so much.  We had another merge in upstream two days ago but we didn't bring it into this repo yet.  Would you mind rebasing onto it if I merged it into master here?  I think README will have an annoying conflict but otherwise it should be easy. There are two options:
```
| `forwardAuth.nameOverride`          | External name of the forward-auth endpoint                                                                                                                                                                                                                                                           | `forwardauth.${rootDomain}`                                                           |
| `forwardAuth.enabled`               | Enable forward-auth endpoint for third party ingress controllers to use for auth checks.  Setting this disables automatic enumeration of `from` hostnames in the Pomerium Ingress object to prevent conflicts.  Use `ingress.hosts` to mix forward-auth and proxy mode on a single Pomerium instance | `false`                                                                               |
```

**Comment by travisgroth** [comment]
Also LGTM otherwise.

**Comment by derrickburns** [comment]
@travisgroth Sure, however I am not clear what you need me to do.  When I pull, I don't see any changes to master.

File: values.yaml [hunk 1]
```diff
@@ -38,6 +38,8 @@ authenticate:
     defaultSANList: []
     defaultIPList: []
   replicaCount: 1
+  service: {}
+  deploymnet: {}
```
Comment (travisgroth) [inline values.yaml]: Typo here.

Also, to pass lint:

```
  service: 
    annotations: {}
  deployment: 
    annotations: {}
```

File: values.yaml [hunk 2]
```diff
@@ -50,6 +52,8 @@ authorize:
...
```
Comment (travisgroth) [inline values.yaml]: See previous comment on linting.

File: values.yaml [hunk 3]
```diff
@@ -65,6 +69,8 @@ proxy:
   authenticateServiceUrl: ""
   authorizeServiceUrl: ""
   authorizeInternalUrl: ""
+  service: {}
+  deploymnet: {}
```
Comment (travisgroth) [inline values.yaml]: See previous comment on linting.

File: values.yaml [hunk 4]
```diff
@@ -38,6 +38,10 @@ authenticate:
     defaultSANList: []
     defaultIPList: []
   replicaCount: 1
+  service:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  service:
```

File: values.yaml [hunk 5]
```diff
@@ -38,6 +38,10 @@ authenticate:
     defaultSANList: []
     defaultIPList: []
   replicaCount: 1
+  service: 
+    annotations: {}
+  deployment:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  deployment:
```

File: values.yaml [hunk 6]
```diff
@@ -50,6 +54,10 @@ authorize:
     defaultSANList: []
     defaultIPList: []
   replicaCount: 1
+  service:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  service:
```

File: values.yaml [hunk 7]
```diff
@@ -50,6 +54,10 @@ authorize:
     defaultSANList: []
     defaultIPList: []
   replicaCount: 1
+  service: 
+    annotations: {}
+  deployment:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  deployment:
```

File: values.yaml [hunk 8]
```diff
@@ -65,6 +73,10 @@ proxy:
   authenticateServiceUrl: ""
   authorizeServiceUrl: ""
   authorizeInternalUrl: ""
+  service:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  service:
```

File: values.yaml [hunk 9]
```diff
@@ -65,6 +73,10 @@ proxy:
   authenticateServiceUrl: ""
   authorizeServiceUrl: ""
   authorizeInternalUrl: ""
+  service: 
+    annotations: {}
+  deployment:
```
Comment (travisgroth) [inline values.yaml]:
```suggestion
  deployment:
```

**Pull request merged**